sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -211,9 +211,10 @@ public ConfigurationClientBuilder credential(ConfigurationClientCredentials cred
      *
      * @param logLevel The amount of logging output when sending and receiving HTTP requests/responses.
      * @return The updated ConfigurationClientBuilder object.
+     * @throws NullPointerException If {@code logLevel} is {@code null}.
      */
     public ConfigurationClientBuilder httpLogDetailLevel(HttpLogDetailLevel logLevel) {
-        httpLogDetailLevel = logLevel;
+        httpLogDetailLevel = Objects.requireNonNull(logLevel);
```

Comment:
What's the justification of not null here? I think make a default one or silence it would be better.

Reply:
Unlike HttpClient and HttpPipeline there is no default handling for HttpLogDetailLevel in the builder. This will end up passing in a null logging level into the logging policy and that will end up throwing a NullPointerException during execution. So add this prevent a consumer from getting further into the call stack to have an error occur.